### PR TITLE
feat: support relational apis

### PR DESCRIPTION
**Pull request by gaurav274**
- [ ] Auto start-server : deferred to another PR
- [x] select
- [x] filter
- [x] load
- [x] create_vector_index
- [x] execute
- [x] order
- [x] limit
- [x] query
- [x] df

### CODE REVIEW

File: eva/interfaces/relational/db.py [hunk 1]
```diff
@@ -23,18 +37,43 @@ def __init__(self, reader, writer):
...
+    ) -> EVARelation:
+        return self.cursor().load(file_regex, table_name, format, **kwargs)
+
+    def table(self, table_name: str) -> EVARelation:
+        return self.cursor().table(table_name)
+
+    def query(self, sql_query: str) -> EVARelation:
+        return self.cursor().query(sql_query)
+
+    def df(self) -> pandas.DataFrame:
```
Comment (jarulraj) [inline eva/interfaces/relational/db.py:59]: Will `self.cursor().query(second_query)` will work even if the first query got a empty result?

File: eva/interfaces/relational/db.py [hunk 2]
```diff
@@ -23,18 +37,43 @@ def __init__(self, reader, writer):
...
+        if not self._result:
+            raise Exception("No valid result with the current connection")
+        return self._result.frames
+
+    def create_vector_index(
+        self, index_name: str, table_name: str, expr: str, using: str
+    ) -> "EVACursor":
```
Comment (jarulraj) [inline eva/interfaces/relational/db.py:66]: Why is this a string instead of a type?

File: eva/interfaces/relational/utils.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        "target_list",
+        "groupby_clause",
+        "orderby_list",
+        "limit_count",
+    ]
+
+    assert clause in accepted_clauses, f"Unknown clause: {clause}"
+
+    # If the clause being set is "target_list" and the value is equal to
+    # "*", the "SELECT *" portion is reset.
```
Comment (jarulraj) [inline eva/interfaces/relational/utils.py:104]: Why is it reset -- may be expand the comment?

**Review by jarulraj — APPROVED** [review]

File: eva/interfaces/relational/db.py (hunk 1 above)
Comment (gaurav274) [reply eva/interfaces/relational/db.py:96]: This is lazy, so you need to call `.df()` on the output to run the query.

File: eva/interfaces/relational/db.py (hunk 2 above)
Comment (gaurav274) [reply eva/interfaces/relational/db.py]: This is a type from the documentation standpoint. This is a way to refer to a class that hasn't been initialized yet. We can change it if you suggest another approach.

File: eva/interfaces/relational/relation.py [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        """Execute and fetch all rows as a pandas DataFrame
+
+        Returns:
+            pandas.DataFrame:
+        """
+        batch = self.execute()
+        assert batch is not None, "relation execute failed"
+        return batch.frames
+
+    def execute(self) -> Batch:
```
Comment (jiashenC) [inline eva/interfaces/relational/relation.py:94]: I understand this function is needed for trigger execution, but I am not sure whether return a `Batch` to users is useful.

Instead, I think providing a `show` function like other systems is more straightforward, which just directly prints results.

Comment (gaurav274) [reply eva/interfaces/relational/relation.py:94]: We have `.df()`. I earlier had `show` as well

File: eva/parser/utils.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
+from eva.parser.load_statement import LoadDataStatement
+
+from eva.parser.parser import Parser
+from eva.parser.select_statement import SelectStatement
+
+
+def parse_expression(expr: str):
+    mock_query = f"SELECT {expr} FROM DUMMY;"
```
[the remark this answers is not in the document]
Comment (gaurav274) [reply eva/parser/utils.py:8]: We don't care about `dummy`. We just need the parsed equivalent of `expr`. The caller takes care of populating i.

File: eva/interfaces/relational/db.py [hunk 6]
```diff
@@ -23,18 +37,43 @@ def __init__(self, reader, writer):
...
 
+    def load(
+        self, file_regex: str, table_name: str, format: str, **kwargs
+    ) -> EVARelation:
+        return self.cursor().load(file_regex, table_name, format, **kwargs)
+
+    def table(self, table_name: str) -> EVARelation:
+        return self.cursor().table(table_name)
+
+    def query(self, sql_query: str) -> EVARelation:
```
[the remark this answers is not in the document]
Comment (gaurav274) [reply eva/interfaces/relational/db.py:84]: Sure 👍

[the remark this answers is not in the document]
Comment (gaurav274) [reply eva/interfaces/relational/db.py:84]: `duckdb` has both. Will do the same

File: eva/interfaces/relational/utils.py (hunk 3 above)
Comment (gaurav274) [reply eva/interfaces/relational/utils.py]: Done!

File: test/interfaces/relational/test_relational_api.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        self.images = f"{EVA_ROOT_DIR}/data/detoxify/*.jpg"
+
+    def tearDown(self):
+        shutdown_ray()
+        # todo: move these to relational apis as well
+        execute_query_fetch_all("""DROP TABLE IF EXISTS mnist_video;""")
+        execute_query_fetch_all("""DROP TABLE IF EXISTS meme_images;""")
+        # os.system("nohup eva_server --stop")
+
+    def test_relation_apis(self):
```
[the remark this answers is not in the document]
Comment (gaurav274) [reply test/interfaces/relational/test_relational_api.py:32]: Done

File: eva/interfaces/relational/db.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (gaurav274) [reply eva/interfaces/relational/db.py:130]: Done

File: eva/interfaces/relational/relation.py (hunk 4 above)
Comment (gaurav274) [reply eva/interfaces/relational/relation.py:94]: Added `show()`

**Pull request merged**